snapcraft/tests/test_plugin_catkin.py

```diff
@@ -0,0 +1,41 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2015 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import fixtures
+
+from snapcraft import tests
+from snapcraft.plugins import catkin
+
+
+class CatkinTestCase(tests.TestCase):
+
+    def setUp(self):
+        super().setUp()
+
+        class props:
+            catkin_packages = ['my_package']
+
+        self.properties = props()
+
+    def test_copy_plugin_copies(self):
```

Comment:
I think you could give this test a better name (or put a comment on what it does). It doesn't test the copy, it tests that a warning is logged if there is no packages.xml.
Maybe something like test_find_package_deps_without_packages_file_must_log_warning.